service/version_metrics.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package service
+
+import (
+	"runtime"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/uber-go/tally"
+)
+
+const reportingTime = 10 * time.Second
+
+type versionMetricsEmitter struct {
+	counter tally.Counter
+	ticker  *time.Ticker
+}
+
+func newVersionMetricsEmitter(scope tally.Scope) *versionMetricsEmitter {
+	t := time.NewTicker(reportingTime)
+	return &versionMetricsEmitter{
+		counter: scope.Tagged(map[string]string{
+			"uberfx-v": fx.Version,
```

Comment:
It's automatically tagged with the service name, since we set up the scope in the manager with the service name. Please see details in the jira ticket :)